service/history/execution/mutable_state_decision_task_manager.go

```diff
@@ -379,8 +379,12 @@ func (m *mutableStateDecisionTaskManagerImpl) AddDecisionTaskScheduledEventAsHea
 	scheduleID := m.msb.GetNextEventID() // we will generate the schedule event later for repeatedly failing decisions
 	// Avoid creating new history events when decisions are continuously failing
 	scheduleTime := m.msb.timeSource.Now().UnixNano()
-	if m.msb.executionInfo.DecisionAttempt == 0 {
-		newDecisionEvent = m.msb.hBuilder.AddDecisionTaskScheduledEvent(taskList, startToCloseTimeoutSeconds,
+	useNonTransientDecision := m.shouldUpdateLastWriteVersion()
+
+	if m.msb.executionInfo.DecisionAttempt == 0 || useNonTransientDecision {
```

Comment:
if `DecisionAttempt` is not 0 here, do we need to reset it to 0?

Reply:
Good catch. yes